### PR TITLE
[QP] Include column aliases on `data.cols` and `result_metadata`

### LINKED ISSUE #43307: [BE] Expose `desired-column-alias` on `result_metadata` and `data.cols` in responses.

### PR DESCRIPTION
**Pull request by bshepherdson**
This is the first part of the branch for #43306.

Fixes #43307.

### CODE REVIEW

**Comment by bshepherdson** [comment]
* **#43790** <a href="https://app.graphite.dev/github/pr/metabase/metabase/43790?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `viz-settings-deref`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bshepherdson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/metabase/lib/metadata/jvm.clj [hunk 1]
```diff
@@ -340,16 +340,19 @@
...
   (case metadata-type
     :metadata/column
     (t2/select :metadata/column
                :table_id        table-id
                :active          true
-               :visibility_type [:not-in #{"sensitive" "retired"}])
-
+               {:where [:or
+                        [:= :field/visibility_type nil]
+                        [:not-in :field/visibility_type #{"sensitive" "retired"}]]})
```
Comment (metamben) [inline src/metabase/lib/metadata/jvm.clj:355]: These look bugfixes, do we have regression tests?

File: src/metabase/query_processor/middleware/annotate.clj [hunk 2]
```diff
@@ -480,23 +480,36 @@
...
+(defn- add-aliases-to-legacy-cols [inner-query cols]
+  (let [returned (lib.metadata.calculation/returned-columns (mlv2-query inner-query))
+        aliases  (into {} (for [col returned]
+                            [[(:name col) (:metabase.lib.join/join-alias col)]
+                             (:lib/desired-column-alias col)]))]
+    (mapv (fn [col]
+            (let [k     [(:name col) (:source_alias col)]
+                  alias (get aliases k)]
+              (cond-> col
+                alias (assoc :desired_column_alias alias))))
```
Comment (metamben) [inline src/metabase/query_processor/middleware/annotate.clj:492]: Isn't this the assoc-some pattern you are advocating?

**Review by metamben — APPROVED** [review]
It seems you've fixed a bunch of bugs here, but these is not obvious when I look at the tests.

File: src/metabase/lib/metadata/jvm.clj (hunk 1 above)
Comment (bshepherdson) [reply src/metabase/lib/metadata/jvm.clj:355]: These are fixes for why `lib.metadata/tables` didn't return anything on JVM (`:not-in` fails for NULL columns). I don't think we actually do that anywhere outside of REPL sessions, though?

File: src/metabase/query_processor/middleware/annotate.clj (hunk 2 above)
Comment (bshepherdson) [reply src/metabase/query_processor/middleware/annotate.clj:492]: Yep, so it is.

File: src/metabase/lib/metadata/jvm.clj (hunk 1 above)
Comment (bshepherdson) [reply src/metabase/lib/metadata/jvm.clj:355]: [This](https://github.com/metabase/metabase/blob/master/test/metabase/lib/test_util.cljc#L243) is the only call anywhere, it seems.

**Pull request merged**